server.go

```diff
@@ -2557,6 +2335,39 @@ func newServer(listenAddrs []string, db database.DB, chainParams *chaincfg.Param
 	}
 	s.cpuMiner = newCPUMiner(&policy, &s)
 
+	// Create a connection manager.
+	connmgrCfg := &connmgr.Config{
+		RetryDuration: connectionRetryInterval,
+		MaxOutbound:   defaultMaxOutbound,
+		Dial:          btcdDial,
+		OnConnection: func(c *connmgr.ConnReq, conn net.Conn) {
+			sp := s.newOutboundPeer(c.Addr, c.Permanent)
+			if sp != nil {
+				if err := sp.Connect(conn); err != nil {
+					srvrLog.Errorf("Can't use connection: %v", err)
+					return
+				}
+				sp.connReq = c
+				srvrLog.Debugf("Connected to %s", sp.Addr())
+				s.addrManager.Attempt(sp.NA())
+			}
+		},
+	}
+	if len(cfg.ConnectPeers) == 0 {
```

Comment:
I'd reverse this a bit so the config is all done in one shot and without the need for the local like so:

```Go
	// Only setup a function to return new addresses to connect to when
	// not running in connect-only mode.
	var newAddressFunc connmgr.AddressFunc
	if len(cfg.ConnectPeers) == 0 {
		newAddressFunc = func() string {
			// ...
		}
	}

	// Create a connection manager.
	cmgr, err := connmgr.New(&connmgr.Config{
		RetryDuration: connectionRetryInterval
		// ...
		GetNewAddress: newAddressFunc,
		Dial:          btcdDial,
	})
	if err != nil {
		return nil, err
	}
```